app/controllers/application_controller.rb

```diff
@@ -1,10 +1,17 @@
 class ApplicationController < ActionController::Base
   before_action :authenticate_user!
   verify_authorized unless: :devise_controller?
-
   before_action :set_current_user
   around_action :switch_locale
 
+  KNOWN_ERRORS = [ActionPolicy::Unauthorized]
+  rescue_from StandardError, with: :log_and_reraise
+  rescue_from ActionPolicy::Unauthorized do |ex|
+    flash[:alert] = t("errors.authorization_error")
+
+    redirect_back fallback_location: root_path
```

Comment:
This is [soft deprecated](https://api.rubyonrails.org/v7.1/classes/ActionController/Redirecting.html#method-i-redirect_back) fyi

```suggestion
    redirect_back_or_to root_path
```